CMakeLists.txt

```diff
@@ -79,6 +79,20 @@ set(CMAKE_VERBOSE_MAKEFILE
     false
     CACHE STRING "Show all outputs including compiler lines." FORCE)
 
+# Force colored output (defaults to ON for Ninja)
+if ("${CMAKE_GENERATOR}" STREQUAL "Ninja")
+  option(FORCE_COLORED_OUTPUT "Always produce ANSI-colored output." ON)
+else ()
+  option(FORCE_COLORED_OUTPUT "Always produce ANSI-colored output." OFF)
+endif ()
+if (${FORCE_COLORED_OUTPUT})
+    if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+       add_compile_options(-fdiagnostics-color=always)
+    elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+       add_compile_options(-fcolor-diagnostics)
+    endif ()
+endif ()
```

Comment:
Afaik CMake has a different compiler id for apple clang. This can be worked around by using `MATCHES` instead of `STREQUL`.
```
option(VAST_COLOR_NINJA "Enable ANSI-colored output for the Ninja generator." ON)
if ("${CMAKE_GENERATOR}" STREQUAL "Ninja" AND VAST_COLOR_NINJA)
  if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
    add_compile_options(-fdiagnostics-color=always)
  elseif ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
    add_compile_options(-fcolor-diagnostics)
  endif ()
endif ()
```